model/task_lifecycle.go

```diff
@@ -817,6 +817,11 @@ func UpdateUnblockedDependencies(t *task.Task) error {
 		if err := UpdateUnblockedDependencies(&blockedTask); err != nil {
 			return errors.WithStack(err)
 		}
+
+		if err := UpdateBuildAndVersionStatusForTask(&blockedTask); err != nil {
```

Comment:
Unit tests?

Reply:
sorry i asked for a review before i pushed the test but they are here now!